### DISCUSSION TITLE
Question about: Automatically populate the QTH "qso_auto_qth"

### DISCUSSION THREAD

**Discussion by m0urs** [magicbug/Cloudlog#1698] [Q&A]
I just found that config item "qso_auto_qth" and as far as I understood it (and the code where this config item is used), it should automatically check for the value in the QTH field and calculate the grid for that location (if the grid field is not already populated).

I set that item to TRUE (both in config/config.php and in the original config/cloudlog.php(, but if I enter a new QSO and enter a value into the QTH field, nothing happens with the grid field value.

Should this function work and did I miss something here?

**Comment by AndreasK79** [comment]
I just tried it in my local install. It works here. I set the value to true in application/config/cloudlog.php.

I enter a city name and tab out of the field. The gridsquare field is now populated in my install.

**Reply by magicbug** [reply]
Yep confirmed same here, actually it needs pulled out and made a frontend changeable option

**Reply by m0urs** [reply]
Hm, ok. Strange. If I do that, a small help screen pops up which tells me, how the gridsquare field should be formated ...

**Reply by m0urs** [reply]
Ah sorry. In the developer tools you find the solution :-(

"Refused to connect to 'https://nominatim.openstreetmap.org/search/?city=frankfurt&format=json&addressdetails=1&limit=1' because it violates the following Content Security Policy directive: "connect-src 'self'"."

Need to add that as allowed in the CSP directive ... My fault.

**Reply by m0urs** [reply]
I confirm: It is now working fo me as well!

**Reply by m0urs** [reply]
> actually it needs pulled out and made a frontend changeable option

I changed it for me in my fork. However, as I am not a real programmer, I am not sure, if I did it the right way. Especially I have no idea how to correctly implement that we need an additional field in the user table in the database. It works however for me, after I manually added that field. Let me know if I still should create a pull request for it.

**Reply by phl0** [reply]
> Yep confirmed same here, actually it needs pulled out and made a frontend changeable option

There is code for that in https://github.com/magicbug/Cloudlog/pull/1705.
